src/datasources/blockchain/blockchain-api.manager.spec.ts

```diff
@@ -79,10 +91,54 @@ describe('BlockchainApiManager', () => {
       configApiMock.getChain.mockResolvedValue(chain);
 
       const api = await target.getApi(chain.chainId);
-      target.destroyApi(chain.chainId);
+      await target.destroyApi(chain.chainId);
       const cachedApi = await target.getApi(chain.chainId);
 
       expect(api).not.toBe(cachedApi);
     });
   });
+
+  describe('createCachedRpcClient', () => {
+    it('caches RPC requests', async () => {
+      const chain = chainBuilder().build();
+      const client = target._createCachedRpcClient(chain);
+      const fetchSpy = jest.spyOn(global, 'fetch');
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      fetchSpy.mockImplementation((_: unknown) => {
+        return Promise.resolve({
+          headers: new Headers({ 'Content-Type': 'application/json' }),
+          ok: true,
+          status: 200,
+          json: () => {
+            // Return chain ID
+            return Promise.resolve({
+              result: faker.string.hexadecimal(),
+            });
+          },
+        } as Response);
+      });
```

Comment:
I'd like feedback on this - ihis goes against how we normally test but I think it's better than mocking viem.

Reply:
I'm OK with spying `fetch` instead of `viem` in this case, I think there's no advantage in one vs. the other, right? Spying `fetch` seems a bit more _e2e-like_. Btw, what do you think about also checking the cache contains the expected key/values here?